### PR TITLE
Add io.httpie.Httpie

### PR DESCRIPTION
**Pull request by flexagoon**
See https://github.com/httpie/desktop/issues/26

### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/httpie/desktop/issues/26#issuecomment-1872004684
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by flexagoon** [comment]
bot, build io.httpie.Httpie

**Comment by flexagoon** [comment]
bot, build io.httpie.Httpie

File: apply_extra.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+# Extract the appimage
+chmod +x httpie.appimage
+unappimage httpie.appimage
+
+# Move all the data to /app/extra/httpie
+mv squashfs-root httpie
+
+# Install icons
+for size in 16x16 32x32 64x64 128x128 256x256 512x512 1024x1024; do
```
Comment (hfiguiere) [inline apply_extra.sh:11]: 1024x1024 is an invalid size.

there is absolutely no point doing this in the apply_extra.

File: io.httpie.Httpie.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      - type: file
+        path: io.httpie.Httpie.metainfo.xml
+
+      # At least a single icon needs to be available at build time to pass the validation.
+      # The rest of the icons are installed later by apply_extra
+      - type: file
+        path: io.httpie.Httpie.png
+    build-commands:
+      - install -D ./apply_extra.sh "${FLATPAK_DEST}/bin/apply_extra"
+      - install httpie "${FLATPAK_DEST}/bin/"
```
Comment (hfiguiere) [inline io.httpie.Httpie.yaml:66]:
```suggestion
      - install -Dm755 httpie "${FLATPAK_DEST}/bin/"
```

File: apply_extra.sh (hunk 1 above)
Comment (flexagoon) [reply apply_extra.sh:11]: > 1024x1024 is an invalid size.

There is a 1024x1024 icon inside of the AppImage. Should it just be ignored?

> there is absolutely no point doing this in the apply_extra.

The icons come from the AppImage, which only gets extracted in apply_extra, so I can't install the icons before that. ([That's how Lunar Client does that as well](https://github.com/flathub/com.lunarclient.LunarClient/blob/72e97269feba8a1835d0845f90e00ba71dad7c14/apply_extra.sh#L19)). How else could I install the icons?

Comment (bbhtt) [reply apply_extra.sh:11]: Add one here (preferably svg) then install it like the desktop file.

File: io.httpie.Httpie.desktop [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Name=HTTPie
+Comment=HTTPie for Desktop
+Exec=httpie %U
+Terminal=false
+Type=Application
+Icon=io.httpie.Httpie
+StartupWMClass=HTTPie
+MimeType=x-scheme-handler/pie;x-scheme-handler/pie;
```
Comment (bbhtt) [inline io.httpie.Httpie.desktop:9]: duplicate?

File: io.httpie.Httpie.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      <li>Work across devices with auto-save and realtime sync</li>
+      <li>Build and preview requests without sending them</li>
+      <li>Use the app offline and without an account</li>
+    </ul>
+  </description>
+  
+  <launchable type="desktop-id">io.httpie.Httpie.desktop</launchable>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://raw.githubusercontent.com/httpie/desktop/master/.github/httpie-desktop.png</image>
```
Comment (bbhtt) [inline io.httpie.Httpie.metainfo.xml:32]: Should be from a tag or commit

File: apply_extra.sh (hunk 1 above)
Comment (flexagoon) [reply apply_extra.sh:11]: I am already doing that with the 256x256 icon:

https://github.com/flexagoon/flathub/blob/647ca282dc321bac6f10cc17cb848b7b9ccf3cd0/io.httpie.Httpie.yaml#L69

The apply_extra script installs the icons in other sizes.

File: io.httpie.Httpie.desktop (hunk 3 above)
Comment (flexagoon) [reply io.httpie.Httpie.desktop:9]: Oops, yes, sorry, that duplicate is in the upstream .desktop file, I forgot to remove it

File: apply_extra.sh (hunk 1 above)
Comment (bbhtt) [reply apply_extra.sh:11]: Do you need icons of every size? If not then install 512px or an svg.

Comment (flexagoon) [reply apply_extra.sh:11]: Upstream doesn't have an svg icon, so I'll use 512px. What's wrong with installing icons in apply_extra though?

Comment (hfiguiere) [reply apply_extra.sh:11]: and it's totally unecesssary because they are not exported

Comment (hfiguiere) [reply apply_extra.sh:11]: remove line 10 to 13.

File: io.httpie.Httpie.yaml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        path: io.httpie.Httpie.desktop
+
+      - type: file
+        path: io.httpie.Httpie.metainfo.xml
+
+      - type: file
+        path: io.httpie.Httpie.png
+    build-commands:
+      - install -D ./apply_extra.sh "${FLATPAK_DEST}/bin/apply_extra"
+      - install -Dm755 httpie "${FLATPAK_DEST}/bin/"
```
Comment (bbhtt) [inline io.httpie.Httpie.yaml:64]:
```suggestion
      - install -Dm755 apply_extra.sh "${FLATPAK_DEST}/bin/apply_extra"
      - install -Dm755 httpie "${FLATPAK_DEST}/bin/httpie"
```

Comment (bbhtt) [reply io.httpie.Httpie.yaml:64]: or use `-t` if you want

**Comment by bbhtt** [comment]
bot, build io.httpie.Httpie

**Comment by flexagoon** [comment]
image

The build seems to work fine

**Comment by bbhtt** [comment]
unappimage seems to output progress and a progress bar that messes up the flatpak cli. Can you redirect that to /dev/null? It does not have a way to silence output it seems.

**Comment by flexagoon** [comment]
Weird, flatpak cli works fine for me here. But sure, I'll silence the output.

**Comment by barthalion** [comment]
It would be nice to explicitly hear back from the upstream team about it.

**Comment by Erick555** [comment]
Does it need to be extra-data? Is httpie appimage non-redistributable?

**Comment by barthalion** [comment]
bot, build io.httpie.Httpie

**Pull request closed without merging**